CHI_TECH/ChiMath/vector_operations.cc

```diff
@@ -0,0 +1,134 @@
+#include "chi_math.h"
+#include <assert.h>
+
+//######################################################### Print Vector
+/** Prints the Vector.*/
+void ChiMath::PrintVector(const VecDbl &x)
+{
+  for (auto& xi : x)
+    std::cout << xi << ' ';
+  std::cout << std::endl;
+}
+
+//#########################################################  Scale
+/** Scales a vector in place by constant.*/
+void ChiMath::Scale(VecDbl &x, const double &val)
+{
+  for (double& xi : x)
+    xi *= val;
+}
+
+//######################################################### Dot Product
+/** Computes the dot product of two vectors.
+ *
+ * \f[
+ * \mathrm{a} \cdot \mathrm{b}=\sum_{i=1}^{n} a_{i} b_{i}
+ * \f]
+ * */
+double ChiMath::Dot(const VecDbl &x, const VecDbl &y)
+{
+  // Error Checks
+  assert(x.size() > 0);
+  assert(y.size() > 0);
+  assert(x.size() == y.size());
+  // Local Variables
+  unsigned int n = x.size();
```

Comment:
This should be size_t.

Reply:
Done. Replaced many references to .size() as unsigned int to size_t